apps/api/src/app/workflows-v2/usecases/get-step-schema/get-step-metadata-use-case.service.ts

```diff
@@ -0,0 +1,150 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { ControlValuesLevelEnum, StepMetadataDto, StepType, StepTypeEnum } from '@novu/shared';
+import { JSONSchema } from 'json-schema-to-ts';
+import { ControlValuesRepository, NotificationStepEntity, NotificationTemplateRepository } from '@novu/dal';
+import { GetStepSchemaCommand } from './get-step-schema.command';
+import { mapStepTypeToResult } from '../../shared';
+
+@Injectable()
+export class GetStepMetadataUseCase {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private controlValuesRepository: ControlValuesRepository
+  ) {}
+
+  async execute(command: GetStepSchemaCommand): Promise<StepMetadataDto> {
+    const { currentStep, previousSteps } = await this.findSteps(command);
+    this.assertStepValidity(currentStep);
+
+    return {
+      controls: {
+        dataSchema: currentStep.controls?.schema,
+        uiSchema: currentStep.controls?.uiSchema,
+        values: await this.getValues(command, currentStep),
+      },
+      variables: buildVariablesSchema(previousSteps),
+    };
+  }
+
+  private async getValues(command: GetStepSchemaCommand, currentStep: NotificationStepEntity) {
+    const controlValuesEntity = await this.controlValuesRepository.findFirst({
+      _environmentId: command.environmentId,
+      _organizationId: command.organizationId,
+      _workflowId: command.workflowId,
+      _stepId: currentStep._templateId,
+      level: ControlValuesLevelEnum.STEP_CONTROLS,
+    });
+
+    return controlValuesEntity?.controls || {};
+  }
+
+  private assertStepValidity(currentStep: NotificationStepEntity) {
+    if (!currentStep.template?.type) {
+      throw new BadRequestException('No step type found');
+    }
+
+    if (!currentStep.template?.controls?.schema) {
+      throw new BadRequestException('No controls schema found');
+    }
```

Comment:
I got this error because the dashboard created steps don't have `template.controls.schema`. 
As I got it right there will be only available `uiSchema`, but I don't see anywhere any logic for that.

![Screenshot 2024-10-31 at 10 38 40](https://github.com/user-attachments/assets/cf7c1ef8-154b-4068-a83d-2d1aa5d2dc0d)

Reply:
talked sync

Reply:
what was the issue?